crates/modelardb_server/src/query/grid_exec.rs

```diff
@@ -373,3 +391,77 @@ impl RecordBatchStream for GridStream {
         self.schema.clone()
     }
 }
+
+/// Metrics collected by [`GridStream`] for use by EXPLAIN ANALYZE.
+#[derive(Debug)]
+struct GridStreamMetrics {
+    /// Default set of metrics collected by operators.
+    baseline_metrics: BaselineMetrics,
+    /// Number of data points reconstructed from segments.
+    created_rows: Count,
+    /// Number of data points reconstructed from segments grouped by model type.
+    created_rows_by_model_type: [Count; 3],
```

Comment:
should this magic number be replaced by a CONSTANT? It is easy to forget to update if/when a new model type is added

Reply:
Fixed by adding `MODEL_TYPE_COUNT`.